internal/util/dependency/kv/kv_client_handler.go

```diff
@@ -0,0 +1,92 @@
+package kvfactory
+
+import (
+	"sync"
+
+	"github.com/tikv/client-go/v2/config"
+	"github.com/tikv/client-go/v2/txnkv"
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	"github.com/milvus-io/milvus/pkg/util/etcd"
+	"github.com/milvus-io/milvus/pkg/util/paramtable"
+)
+
+// ETCDReuser is mainly for sessions. This function allows the etcd client to be made after all the configs
+// have been set and allows for the reuise of the etcd client for a package.
+type ETCDReuser struct {
```

Comment:
It should be private.

The naming is probably not accurate. maybe `etcdClientCreator`?